### PR TITLE
Initial version of librmm recipe for conda-forge

### PR DESCRIPTION
**Pull request by rlratzel**
This is an initial version of a librmm recipe for conda-forge.

librmm is a C++/CUDA header-only library for the RAPIDS Memory Manager, described [here](https://github.com/rapidsai/rmm).

The above linked repo also includes a python interface known as simply `rmm`, but this recipe is only for the librmm component, to satisfy the new `libcugraph` CuPy dependency.

Notes:
* I'm listed as the maintainer and this is my comment confirming I'm willing to be listed there. @kkraus14 verbally requested to be added as a maintainer too, and based on [this comment](https://github.com/conda-forge/libcugraph-feedstock/pull/7#issuecomment-824977102), I'm assuming @leofang wants to be a maintainer as well.
* This is a **header-only** C++ and CUDA package, so I will contact @conda-forge/help-c-cpp for help if necessary. cc @jakirkham

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jakirkham** [comment]
Thanks Rick! 😄

Feel free to add me as a maintainer as well 🙂

File: recipes/librmm/meta.yaml
Comment (jakirkham) [inline recipes/librmm/meta.yaml:15]: Should skip the CPU build, which is currently failing on CI

```suggestion
  skip: true  # [(not linux) or (cuda_compiler_version == "None")]
```

File: recipes/librmm/meta.yaml
Comment (jakirkham) [inline recipes/librmm/meta.yaml:27]: Should this be a `run` requirement as well?

Comment (rlratzel) [reply recipes/librmm/meta.yaml:27]: I believe we're using it as a header-only library, but @kkraus14 can confirm.

File: recipes/librmm/meta.yaml
Comment (rlratzel) [reply recipes/librmm/meta.yaml]: Thanks John, fixed in next commit.

File: recipes/librmm/meta.yaml
Comment (kkraus14) [reply recipes/librmm/meta.yaml:27]: Yes it should be: https://github.com/rapidsai/rmm/blob/branch-0.20/conda/recipes/librmm/meta.yaml#L38

Comment (jakirkham) [reply recipes/librmm/meta.yaml:27]: Thanks for the clarification. So it should be `run` only. Included a code suggestion below...

File: recipes/librmm/meta.yaml
Comment (jakirkham) [inline recipes/librmm/meta.yaml:27]:
```suggestion
  run:
    - spdlog =1.7.0
```

Comment (rlratzel) [reply recipes/librmm/meta.yaml:27]: I see, this installs it for the `librmm` clients, since they'll need it when they build with `librmm`.  I'll add this in the next commit.

File: recipes/librmm/meta.yaml
Comment (rlratzel) [reply recipes/librmm/meta.yaml:27]: Thanks, fixed in https://github.com/conda-forge/staged-recipes/pull/14749/commits/ae4496c604ee1642817aebf1b1d47ed2f9bfa4bb

File: recipes/librmm/meta.yaml
Comment (jakirkham) [reply recipes/librmm/meta.yaml]: Yeah so this should ensure `cupy` still has `spdlog` during the build

File: recipes/librmm/meta.yaml
Comment (kkraus14) [inline recipes/librmm/meta.yaml:24]: I think this should actually be `>=3.18.0`, we recently fixed this in 0.20.
```suggestion
    - cmake >=3.18
```

File: recipes/librmm/meta.yaml
Comment (kkraus14) [inline recipes/librmm/meta.yaml:26]: We also need to include thrust here as we leak the thrust dependency: https://github.com/rapidsai/rmm/blob/branch-0.20/cmake/Modules/RMM_thirdparty.cmake

Comment (rlratzel) [reply recipes/librmm/meta.yaml:26]: Added in next commit. I used `>=1.9.0` based on the min version in the linked file.

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
LGTM, once @jakirkham signs off lets merge 😄

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Rick for working on this PR! Also thanks for the review Keith 😄

Let's handle anything else in the feedstock 🙂

File: recipes/librmm/meta.yaml
Comment (leofang) [reply recipes/librmm/meta.yaml:26]: @kkraus14 Does this mean all downstream packages of librmm are forced to use Conda-Forge's Thrust/CUB?

**Pull request merged**
